src/checks/hyper-training-6iv.js

```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  description: 'A 6IV Pokémon cannot be hyper trained.',
+  filter: {ivHp: 31, ivAtk: 31, ivDef: 31, ivSpAtk: 31, ivSpDef: 31, ivSpe: 31},
+  field: 'iv',
+  test (pkmn) {
+    return !(pkmn.hyperTrainedHp || pkmn.hyperTrainedAtk ||
+        pkmn.hyperTrainedDef || pkmn.hyperTrainedSpAtk ||
+        pkmn.hyperTrainedSpDef || pkmn.hyperTrainedSpe);
+  }
+};
```

Comment:
This check seems redundant with hyper-training-31. Anything that it marks illegal will also be marked illegal by that other check.